cantor-s3/src/main/java/com/salesforce/cantor/s3/utils/S3Utils.java

```diff
@@ -0,0 +1,183 @@
+package com.salesforce.cantor.s3.utils;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.ObjectListing;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+import com.salesforce.cantor.s3.ObjectsOnS3;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+public class S3Utils {
+    private static final Logger logger = LoggerFactory.getLogger(ObjectsOnS3.class);
+
+    // read objects in 4MB chunks
+    private static final int streamingChunkSize = 4 * 1024 * 1024;
+
+    public static void createBucket(final AmazonS3 s3Client, final String bucketName) throws IOException {
+        if (s3Client.doesBucketExistV2(bucketName)) {
+            logger.info("bucket '{}' already exists; ignoring create", bucketName);
+            return;
+        }
+        logger.info("bucket '{}' doesn't exist; creating it", bucketName);
+        s3Client.createBucket(bucketName);
+
+        // check bucket created successfully
+        if (!s3Client.doesBucketExistV2(bucketName)) {
+            throw new IOException("failed to create namespace on s3 with bucket name: " + bucketName);
+        }
+    }
+
+    public static void deleteBucket(final AmazonS3 s3Client, final String bucketName) throws IOException {
+        deleteObjects(s3Client, bucketName, null);
+
+        // now delete the bucket
+        s3Client.deleteBucket(bucketName);
+
+        // check bucket deleted successfully
+        if (s3Client.doesBucketExistV2(bucketName)) {
+            throw new IOException("failed to drop namespace on s3 with bucket name: " + bucketName);
+        }
+    }
+
+    public static Collection<String> getKeys(final AmazonS3 s3Client,
+                                             final String key,
+                                             final String prefix,
+                                             final int start,
+                                             final int count) throws IOException {
+        if (!s3Client.doesBucketExistV2(key)) {
+            throw new IOException(String.format("couldn't find bucket '%s'", key));
+        }
+
+        final Set<String> keys = new HashSet<>();
+        int index = 0;
+        ObjectListing listing = s3Client.listObjects(key, prefix);
+        do {
+            for (final S3ObjectSummary summary : listing.getObjectSummaries()) {
+                if (index < start) {
```

Comment:
@AlexanderKouthoofd instead of iterating over elements and skip until `start`, let's short circuit this if length is smaller than start.